### PR TITLE
Make the demoRecorder write the experience on reset

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

Fixing the demo recording of experiences when the agent is done

### Useful links (Github issues, JIRA tickets, ML-Agents forum threads etc.)

ML-640

### Types of change(s)

- [x] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added updated the changelog (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] I have updated the migration guide (if applicable)

### CODE REVIEW

**Comment by ervteng** [comment]
BTW can you re-record `test.demo` (it's just 3DBall using inference) in the `test` directory? Just so we can make sure the Python is compatible with the new demos

File: com.unity.ml-agents/Runtime/Agent.cs [hunk 1]
```diff
@@ -702,16 +707,25 @@ void AgentStep()
...
-                {
-                    AgentAction(m_Action.vectorActions);
-                }
+                AgentAction(m_Action.vectorActions);
             }
         }
 
         void DecideAction()
         {
             m_Action.vectorActions = m_Brain?.DecideAction();
```
Comment (ervteng) [inline com.unity.ml-agents/Runtime/Agent.cs]: I'm guessing this is set to `null` on reset?

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Agent.cs]: Yeah, I think that might be better; ResetData() resets the stored actions, but this change keeps them (possibly with stale data).

Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/Agent.cs]: We are not really reseting the data, we are generating a vector of zeros in the case the action is null

Comment (vincentpierre) [reply com.unity.ml-agents/Runtime/Agent.cs]: Alright, changed it since there seems to be consensus.

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Agent.cs]: Then how about moving the vectorAction code from ResetData to a new method (ResetActions()?), call that from ResetData, and call it here?

Comment (chriselion) [reply com.unity.ml-agents/Runtime/Agent.cs]: Sorry, missed your most recent comment before I made mine.

**Pull request merged**